Clinical-Genomics/scout
# De-archive re-runs and sorting of cases page

This PR adds the following functionalities:
- Cases that are **re-run become 'inactive'** if they were previously archived (fix #1271)
- Makes it possible to **sort cases by 'Status', 'Analysis date' and 'Track'** in cases page (fix #1271)
- Orders the categories of displayed cases (cases page) like this: **'prioritized', 'active', 'inactive', 'archived', 'solved'** (fix #1285)


**How to test dearchiving of re-runs**:
1. On a local installation of scout-stage, archive the test case
1. From the command line update the case with the following command:
```
 scout --demo load case scout/demo/643594.config.yaml --update
```
3. Expected outcome: Refresh the page and make sure that the case status is now 'inactive'


**How to test the case sorting by 'Status', 'Analysis date' or 'Track' :**
1. Install the branch on stage with the following commands:
```
ssh your.user@clinical-db.scilifelab.se
sudo -iu hiseq.clinical
cd servers/resources/clinical-dbscilifelab.se
bash update-scout-stage.sh dearchive_reruns
```
2. Go to 'cases' web page of scout-stage and make sure that you see sorting icons near the 3 fields:
![image](https://user-images.githubusercontent.com/28093618/63768440-7d276f80-c8d0-11e9-8c22-e840930c45e7.png)
Play with the sorting commands. Does the sorting make sense? Tip: Limit the number of cases to 1000 instead of 100


**How to test the status category order in case page:**
1. After installing the branch on stage (already done in previous step) go to cases page and make sure that the categories displayed follow **always this order: 'prioritized', 'active', 'inactive', 'archived', 'solved'** . Compare with the display order of scout production and you'll see that the order is different. 
1. Note that the status category order might be affected if you choose to sort cases according to status, analysis_date or track if you are limiting the cases displayed to 100 (or less). One could always remove the limit option when using this type of sorting..

**Review:**
- [x ] code approved by dNil
